RecaptchaValidation/Interfaces/IRecaptchaService.cs

```diff
@@ -1,10 +1,10 @@
-﻿namespace RecaptchaValidation.Interfaces
+﻿using System.Threading.Tasks;
+using RecaptchaValidation.Models;
+
+namespace RecaptchaValidation.Interfaces
 {
     public interface IRecaptchaService
     {
-        IRecaptchaRequestMessage Request { get; set; }
-        IRecaptchaResponseMessage Response { get; set; }
-        void InitializeRequest(IRecaptchaRequestMessage request);
-        Task<IRecaptchaResponseMessage> Execute();
+        Task<RecaptchaResponseMessage> Execute(RecaptchaRequestMessage requestMessage);
```

Comment:
Typically convention says to add the `Async` suffix to any function that performs an asynchronous action. So this could be called `ExecuteAsync`.

This is losing favor in modern days, but it is a decent convention for people to follow in my opinion.

Also `Execute` is kind of a lacklust name. A bit meaningless. Another name could be `ValidateRecaptchaAsync` or something of the like.

Reply:
`Execute()` is the name of the function that is used in the front end to kick this whole process off and in general seems to be a pretty well understood name within the context of Google's Recaptcha process. I'll meet you in the middle and go with `ExecuteAsync()`